### ISSUE TITLE
Parser should accept 1 and 0 as boolean literals in MySQL

### ISSUE THREAD

**Issue by vmenshov** [jOOQ/jOOQ#12394]
I use parser to create canonical form of MySql 5.7 queries, for statistical purposes, and sometimes i need to parse queries like this one:

```sql
select name from users where 1 and id = 500;
```

Such queries are valid for the MySql 5.7 database, but the JOOQ 3.15.2 open source edition throws an exception while trying to parse it.

I receive:
```
Exception in thread "main" org.jooq.impl.ParserException: Boolean field expected: [1:36] select name from users where 1 and [*]id = 500;
	at org.jooq.impl.ParserContext.expected(ParserImpl.java:10641)
	at org.jooq.impl.ParserImpl.toCondition(ParserImpl.java:5258)
	at org.jooq.impl.ParserImpl.parseAnd(ParserImpl.java:4343)
       ...............................
```
Code to reproduce:

```java
import org.jooq.*;import org.jooq.conf.*;
import org.jooq.impl.DSL;
import org.jooq.impl.DefaultConfiguration;

public class MainClass {
    public static void main(String[] args) {
        DSLContext ctx = DSL.using(
                new DefaultConfiguration().set(
                        new Settings()
                                .withParamType(ParamType.FORCE_INDEXED)
                                .withParseDialect(SQLDialect.MYSQL)
                                .withRenderNameCase(RenderNameCase.LOWER)
                                .withRenderKeywordCase(RenderKeywordCase.LOWER)
                                .withRenderFormatted(true)
                                .withRenderQuotedNames(RenderQuotedNames.NEVER)
                                .withParseUnsupportedSyntax(ParseUnsupportedSyntax.IGNORE)
                                .withParseUnknownFunctions(ParseUnknownFunctions.IGNORE)
                                .withThrowExceptions(ThrowExceptions.THROW_NONE)));
        Parser parser = ctx.parser();
        Queries queries = parser.parse("select name from users where 1 and id = 500;");
        Query query = queries.queries()[0];
        System.out.println(String.format("parsed query: %s%nparams: %s", query, query.getParams()));
    }
}
```

Is it possible to fix parser in open source edition?

**Comment by lukaseder** [comment]
Thanks a lot for your report. It's probably non-trivial to get this right for all types of expressions, especially when the parser does not know the data type of an expression, but in this particular case, when people use `1` or `0` as "boolean literals", we could probably make an exception. That might already cover half of these cases for people using this "feature" of MySQL.

> Is it possible to fix parser in open source edition?

You can do whatever you want with the jOOQ Open Source Edition

**Comment by lukaseder** [comment]
An interesting question that arises from such a change would be long term effects on operator precedence. E.g. this is possible in MySQL:

```sql
SELECT 1 = 1 = 1
```

The same thing is not possible in PostgreSQL:

```sql
-- Fails
select true = true = true

-- Works
select (true = true) = true
```

Parentheses are required in PG

**Comment by lukaseder** [comment]
That latter thing will be fixed separately: https://github.com/jOOQ/jOOQ/issues/12396

**Comment by lukaseder** [comment]
Might as well get this right. All numeric values other than `0` behave as `true`. A quick confirmation:

```sql
with t (a, b, c) as (
  values 
    row (2, 1, -1),
    row (2, 1, 0), 
    row (2, 0, -1),
    row (2, 0, 0),
    row (0, 1, -1),
    row (0, 1, 0), 
    row (0, 0, -1),
    row (0, 0, 0)
)
select a, b, c, a or b or c
from t;
```

Resulting in:

```
|a  |b  |c  |a or b or c|
|---|---|---|-----------|
|2  |1  |-1 |1          |
|2  |1  |0  |1          |
|2  |0  |-1 |1          |
|2  |0  |0  |1          |
|0  |1  |-1 |1          |
|0  |1  |0  |1          |
|0  |0  |-1 |1          |
|0  |0  |0  |0          |
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.3 (https://github.com/jOOQ/jOOQ/issues/12397). Thanks again for your report.

**Comment by vmenshov** [comment]
Thank you!
